src/main/java/com/training/springpractice/controllers/CatalogController.java

```diff
@@ -0,0 +1,64 @@
+package com.training.springpractice.controllers;
+
+import com.training.springpractice.models.Catalog;
+import com.training.springpractice.models.CatalogBody;
+import com.training.springpractice.services.CatalogService;
+import org.modelmapper.ModelMapper;
+import org.modelmapper.PropertyMap;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+
+@RestController
+@RequestMapping("/api/v1/catalog-entry")
+public class CatalogController {
+
+    @Autowired
+    private CatalogService service;
```

Comment:
The injection could be done in different ways, here you are using Field Injection.   Could you explain me please why you decided to use this type of injection?
There is the constructor type injection, what are the advantages of using one or another type of injection?
And finally which one is the most recommended way of using dependency injection?